### PR TITLE
chore(maps): Upgrade lifecycle for GeoDistanceSlider

### PR DESCRIPTION
**Pull request by jyash97**
GDS

### CODE REVIEW

File: packages/maps/src/components/basic/GeoDistanceSlider.js [hunk 1]
```diff
@@ -52,71 +45,68 @@ class GeoDistanceSlider extends Component {
...
-				},
-			);
+		if (props.selectedValue && props.selectedValue.location) {
+			currentLocation = props.selectedValue.location;
+			this.getCoordinates(props.selectedValue.location, () => {
+				if (props.selectedValue.distance) {
+					this.setDistance(props.selectedValue.distance);
+				}
+			});
+		} else if (props.defaultSelected && props.defaultSelected.location) {
```
Comment (metagrover) [inline packages/maps/src/components/basic/GeoDistanceSlider.js]: we should remove `defaultSelected` and use `value` - `defaultValue` props.

File: packages/maps/src/components/basic/GeoDistanceSlider.js [hunk 2]
```diff
@@ -52,71 +45,68 @@ class GeoDistanceSlider extends Component {
...
+	componentDidUpdate(prevProps) {
+		checkPropChange(this.props.react, prevProps.react, () => this.setReact(this.props));
 
-		checkSomePropChange(this.props, nextProps, ['dataField', 'nestedField'], () => {
-			this.updateQuery(this.state.currentDistance, nextProps);
+		checkSomePropChange(this.props, prevProps, ['dataField', 'nestedField'], () => {
+			this.updateQuery(this.state.currentDistance, this.props);
 		});
 
 		if (
```
Comment (metagrover) [inline packages/maps/src/components/basic/GeoDistanceSlider.js:91]: same here - watch for changes on `value` prop if defined.

**Review by metagrover — CHANGES_REQUESTED** [review]
We need to cover controlled/uncontrolled component use-case as well.

File: packages/maps/src/components/basic/GeoDistanceSlider.js (hunk 1 above)
Comment (jyash97) [reply packages/maps/src/components/basic/GeoDistanceSlider.js]: Have done defaultValue changes but wasnt able to make it control.

**Comment by jyash97** [comment]
I have done this changes in #918

**Pull request closed without merging**